### PR TITLE
Update NativeCallable attribute tests

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
Fix tests for the NativeCallableAttribute. #18906 

cc @jkotas @luqunl

### CODE REVIEW

File: tests/src/Interop/NativeCallable/NativeCallableTest.cs [hunk 1]
```diff
@@ -17,148 +18,240 @@ public class Program
...
+        catch (NotSupportedException)
         {
-            Console.WriteLine(":bla");
         }
     }
 
+    [NativeCallable]
+    public static int CallbackMethodGeneric<T>(T arg)
+    {
+        Assert.Fail($"Functions with attribute {nameof(NativeCallableAttribute)} cannot have generic aruments");
```
Comment (jkotas) [inline tests/src/Interop/NativeCallable/NativeCallableTest.cs]: Typo: aruments

File: tests/src/Interop/NativeCallable/NativeCallableTest.cs [hunk 2]
```diff
@@ -17,148 +18,240 @@ public class Program
...
+        {
+            Func<int, int> invoker = ManagedDoubleCallback;
+            invoker(0);
         }
     }
 
+    [NativeCallable]
+    public static int CallbackMethodNonBlittable(bool x1)
+    {
+        Assert.Fail($"Functions with attribute {nameof(NativeCallableAttribute)} cannot have non-blittable aruments");
```
Comment (jkotas) [inline tests/src/Interop/NativeCallable/NativeCallableTest.cs]: Typo: aruments

**Comment by AaronRobinsonMSFT** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test

**Comment by AaronRobinsonMSFT** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test

**Comment by AaronRobinsonMSFT** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test

**Comment by AaronRobinsonMSFT** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test

**Comment by AaronRobinsonMSFT** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test

**Pull request merged**